Source/Installer/AppDelegate.h

```diff
@@ -34,20 +34,20 @@
     ArchiveUtil *_archiveUtil;
     NSString *_installingVersion;
     BOOL _upgrading;
-    NSButton *_installButton;
-    NSButton *_cancelButton;
-    NSTextView *_textView;
-    NSWindow *_progressSheet;
-    NSProgressIndicator *_progressIndicator;
+    NSButton *__weak _installButton;
+    NSButton *__weak _cancelButton;
+    NSTextView *__unsafe_unretained _textView;
+    NSWindow *__weak _progressSheet;
+    NSProgressIndicator *__weak _progressIndicator;
     NSDate *_translocationRemovalStartTime;
     NSInteger _currentVersionNumber;
 }
 - (IBAction)agreeAndInstallAction:(id)sender;
 - (IBAction)cancelAction:(id)sender;
 
-@property (assign) IBOutlet NSButton *installButton;
-@property (assign) IBOutlet NSButton *cancelButton;    
-@property (assign) IBOutlet NSTextView *textView;
-@property (assign) IBOutlet NSWindow *progressSheet;
-@property (assign) IBOutlet NSProgressIndicator *progressIndicator;
+@property (weak) IBOutlet NSButton *installButton;
+@property (weak) IBOutlet NSButton *cancelButton;    
+@property (unsafe_unretained) IBOutlet NSTextView *textView;
```

Comment:
Hmm I haven't done this exercise for a long time :/ any reason why `textView` can't be `weak` like others? (Same question for line 39.)

Reply:
See https://developer.apple.com/library/archive/releasenotes/ObjectiveC/RN-TransitioningToARC/Introduction/Introduction.html#//apple_ref/doc/uid/TP40011226

    Which classes don’t support weak references?

    You cannot currently create weak references to instances of the following classes:

    NSATSTypesetter, NSColorSpace, NSFont, NSMenuView, NSParagraphStyle, NSSimpleHorizontalTypesetter, and NSTextView.

Reply:
That is what happens if we change it to weak.

![Screen Shot 2021-11-11 at 00 11 24](https://user-images.githubusercontent.com/53011/141150186-e2c53e84-8c48-4ed6-bae5-2c75ea1b2a97.png)